numpy/core/tests/test_multiarray.py

```diff
@@ -89,14 +89,23 @@ def test_writeable(self):
 
     def test_otherflags(self):
         assert_equal(self.a.flags.carray, True)
+        assert_equal(self.a.flags['C'], True)
         assert_equal(self.a.flags.farray, False)
         assert_equal(self.a.flags.behaved, True)
         assert_equal(self.a.flags.fnc, False)
         assert_equal(self.a.flags.forc, True)
         assert_equal(self.a.flags.owndata, True)
         assert_equal(self.a.flags.writeable, True)
         assert_equal(self.a.flags.aligned, True)
-        assert_equal(self.a.flags.updateifcopy, False)
+        with warnings.catch_warnings(record=True) as sup:
+            warnings.filterwarnings('always', '', DeprecationWarning)
+            warnings.filterwarnings('always', '', FutureWarning)
+            assert_equal(self.a.flags.updateifcopy, False)
+            assert_equal(self.a.flags['U'], False)
+            assert len(sup) == 2
+        assert_equal(self.a.flags.writebackifcopy, False)
+        assert_equal(self.a.flags['X'], False)
```

Comment:
Can you use `numpy.testing.assert_warns` here?

Reply:
then I should wrap each test in a ``assert_warns`` context manager? (likewise line 4323)

Reply:
Something like that? Honestly I don't remember the details, I just remember that @seberg had to go on like some ridiculous epic quest in order to get reliable warning tests working, so we should make use of his work :-). (@seberg, any hints?)

Reply:
Nothing special there, the epic quest is just to never use "ignore" filter (and we have a test to make sure you don't do that). Assert warns as context filter is great, if you want to use the message text, you can use `suppress_warnings` directly.